tendrl/gluster_integration/objects/volume/atoms/check_volume_available/__init__.py

```diff
@@ -0,0 +1,52 @@
+import etcd
+import gevent
+import subprocess
+
+from tendrl.commons.event import Event
+from tendrl.commons.message import Message
+from tendrl.commons import objects
+from tendrl.commons.objects import AtomExecutionFailedError
+from tendrl.gluster_integration.objects.volume import Volume
+
+
+class CheckVolumeAvailable(objects.BaseAtom):
+    def __init__(self, *args, **kwargs):
+        super(CheckVolumeAvailable, self).__init__(*args, **kwargs)
+
+    def run(self):
+        retry_count = 0
+        while True:
+            try:
+                volumes = NS._int.client.read(
```

Comment:
use object.load() , we need to ensure all attributes inside the object are also present as provided by user

Because a Volume is only really fully available when you can load() it


Also, just load the actual volume instead of iterating over every volume